### PR TITLE
[API-DRAFT]Add spark-core-starter module Code splitting for old and new engines

### PR DESCRIPTION
**Pull request by CalvinKirs**
Add `spark-core-starter` module Code splitting for old and new engines

### CODE REVIEW

File: seatunnel-core/seatunnel-spark-starter/src/main/bin/start-seatunnel-spark.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+fi
+
+if [ $# == 0 ]
+then
+    args="-h"
+else
+    args=$@
+fi
+
+CMD=$(java -cp ${APP_JAR} org.apache.seatunnel.core.spark.SparkStarter ${args} | tail -n 1) && EXIT_CODE=$? || EXIT_CODE=$?
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-spark-starter/src/main/bin/start-seatunnel-spark.sh]:
```suggestion
CMD=$(java -cp ${APP_JAR} org.apache.seatunnel.core.starter.spark.SparkStarter ${args} | tail -n 1) && EXIT_CODE=$? || EXIT_CODE=$?
```

File: seatunnel-core/seatunnel-spark-starter/src/test/resources/spark_application.conf [hunk 2]
```diff
@@ -0,0 +1,66 @@
+#
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-spark-starter/src/test/resources/spark_application.conf:1]: It seems this filed have not been used?

File: seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/config/SparkExecutionContext.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        this.sparkSinkPluginDiscovery = new SparkSinkPluginDiscovery();
+        List<URL> pluginJars = new ArrayList<>();
+        pluginJars.addAll(sparkSourcePluginDiscovery.getPluginJarPaths(getPluginIdentifiers(PluginType.SOURCE)));
+        pluginJars.addAll(sparkSinkPluginDiscovery.getPluginJarPaths(getPluginIdentifiers(PluginType.SINK)));
+        this.pluginJars = pluginJars;
+        this.getEnvironment().registerPlugin(pluginJars);
+    }
+
+    @Override
+    public List<BaseSource<SparkEnvironment>> getSources() {
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/config/SparkExecutionContext.java]: In our new spark starter module we will not use this class, BaseSource is used for old API.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: seatunnel-core/seatunnel-spark-starter/src/test/resources/spark_application.conf (hunk 2 above)
Comment (CalvinKirs) [reply seatunnel-core/seatunnel-spark-starter/src/test/resources/spark_application.conf:1]: used in test

Comment (CalvinKirs) [reply seatunnel-core/seatunnel-spark-starter/src/test/resources/spark_application.conf:1]: used in test~

**Review by ruanwenjun — APPROVED** [review]

**Pull request merged**